counterexamples/char_p_zero_ne_char_zero.lean

```diff
@@ -0,0 +1,25 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa, Eric Wieser
+-/
+import algebra.char_p.basic
+
+/-! # `char_p R 0` and `char_zero R` need not coincide for semirings
+
+This file shows that there are semiring `R` for which `char_p R 0` holds and `char_zero R` does not.
```

Comment:
Might be worth to briefly point out that for rings everything is fine. Just for the peace of mind of the casual reader.

Reply:
Thanks Johan: I added the comment.  While looking into this, I also weakened the assumptions on one of the relevant lemmas:  #13214.